app/controllers/assignment_invitations_controller.rb

```diff
@@ -8,19 +9,79 @@ class AssignmentInvitationsController < ApplicationController
   before_action :ensure_submission_repository_exists, only: %i[setup setup_progress success]
   before_action :ensure_authorized_repo_setup, only: %i[setup setup_progress]
 
+  # rubocop:disable PerceivedComplexity
+  # rubocop:disable MethodLength
+  # rubocop:disable AbcSize
+  # rubocop:disable CyclomaticComplexity
   def accept
-    create_submission do
-      GitHubClassroom.statsd.increment("exercise_invitation.accept")
-      if current_submission.starter_code_repo_id
-        redirect_to setup_assignment_invitation_path
-      else
-        redirect_to success_assignment_invitation_path
+    if import_resiliency_enabled?
+      result = current_invitation.redeem_for(current_user, import_resiliency: import_resiliency_enabled?)
+      case result.status
+      when :success
+        GitHubClassroom.statsd.increment("exercise_invitation.accept")
+        if current_submission.starter_code_repo_id
+          redirect_to setup_assignment_invitation_path
+        else
+          current_invitation.completed!
+          redirect_to success_assignment_invitation_path
+        end
+      when :pending
+        GitHubClassroom.statsd.increment("exercise_invitation.accept")
+        redirect_to setupv2_assignment_invitation_path
+      when :error
+        GitHubClassroom.statsd.increment("exercise_invitation.fail")
+        current_invitation.errored!
+
+        flash[:error] = result.error
+        redirect_to assignment_invitation_path(current_invitation)
+      end
+    else
+      create_submission do
+        GitHubClassroom.statsd.increment("exercise_invitation.accept")
+        if current_submission.starter_code_repo_id
+          redirect_to setup_assignment_invitation_path
+        else
+          redirect_to success_assignment_invitation_path
+        end
       end
     end
   end
+  # rubocop:enable PerceivedComplexity
+  # rubocop:enable MethodLength
+  # rubocop:enable AbcSize
+  # rubocop:enable CyclomaticComplexity
 
   def setup; end
 
+  def setupv2
+    render status: 404 unless import_resiliency_enabled?
+  end
+
+  # rubocop:disable MethodLength
+  def create
+    if import_resiliency_enabled?
+      job_started = false
+      if current_invitation.accepted? || current_invitation.errored?
+        AssignmentRepo::CreateGitHubRepositoryJob.perform_later(current_assignment, current_user)
```

Comment:
Should we ensure that the job is a unique job before we kick it off? Since jobs retry with exponential backoff we could end up in a situation where another job is kicked off while another is about to be retired.

Reply:
By default Sidekiq doesn't have this feature. It does in enterprise, but we aren't on that plan.

Another option would be to use [mhenrixon/sidekiq-unique-jobs](https://github.com/mhenrixon/sidekiq-unique-jobs). This would require us to move away from ActiveJob and to using straight Sidekiq.

Possibly worth investigating in a follow up PR.